### PR TITLE
Add support of sorting dictionary of other primitive arrays

### LINKED ISSUE #2700: Add support of sorting dictionary of other primitive types
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

Add the support of sorting dictionary array of remaining primitive types except for integers

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by viirya**
# Which issue does this PR close?

Closes #2700.

# Rationale for this change
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: arrow/src/compute/kernels/sort.rs [hunk 1]
```diff
@@ -314,107 +314,183 @@ pub fn sort_to_indices(
...
-                            !options.nulls_first
-                        } else {
-                            options.nulls_first
-                        };
-                        let value_options = Some(SortOptions { descending: false, nulls_first: value_null_first });
+                        let sorted_value_indices = sort_to_indices(dict_values, value_options, None)?;
+                        let value_indices_map = prepare_indices_map(&sorted_value_indices);
+                        sort_primitive_dictionary::<_, _>(values, &value_indices_map, v, n, options, limit, cmp)
+                    },
+                    DataType::Float32 => {
```
Comment (tustvold) [inline arrow/src/compute/kernels/sort.rs:384]: Perhaps we could collapse these match statements together, they appear to all be identical?

Comment (viirya) [reply arrow/src/compute/kernels/sort.rs]: Yea, forgot that this matches value types. 😅

**Review by tustvold — APPROVED** [review]

File: arrow/src/compute/kernels/sort.rs [hunk 2]
```diff
@@ -314,119 +314,40 @@ pub fn sort_to_indices(
...
-                        } else {
-                            options.nulls_first
-                        };
-                        let value_options = Some(SortOptions { descending: false, nulls_first: value_null_first });
-                        let sorted_value_indices = sort_to_indices(dict_values, value_options, None)?;
-                        let value_indices_map = prepare_indices_map(&sorted_value_indices);
-                        sort_primitive_dictionary::<_, _>(values, &value_indices_map, v, n, options, limit, cmp)
-                    },
-                    DataType::Int16 => {
+                    DataType::Int8 | DataType::Int16 | DataType::Int32 | DataType::Int64 | DataType::UInt8 |
```
Comment (tustvold) [inline arrow/src/compute/kernels/sort.rs:335]: I wonder if this should be a member function somewhere...

File: arrow/src/compute/kernels/sort.rs [hunk 3]
```diff
@@ -314,119 +314,40 @@ pub fn sort_to_indices(
...
+                    DataType::UInt16 | DataType::UInt32 | DataType::UInt64 | DataType::Float32 | DataType::Float64 |
+                    DataType::Date32 | DataType::Date64 | DataType::Time32(Second) | DataType::Time32(Millisecond) |
+                    DataType::Time64(Microsecond) | DataType::Time64(Nanosecond) | DataType::Timestamp(Second, _) |
```
Comment (tustvold) [inline arrow/src/compute/kernels/sort.rs:338]: Do we need to match the specific TimeUnits?

Comment (viirya) [reply arrow/src/compute/kernels/sort.rs]: No, I think. Just copied from above match statements. Removed them now.

File: arrow/src/compute/kernels/sort.rs (hunk 2 above)
Comment (viirya) [reply arrow/src/compute/kernels/sort.rs]: Added one helper function for that.

**Comment by viirya** [comment]
Thanks.

**Pull request merged**